crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -4774,17 +4787,13 @@ impl<'a> Checker<'a> {
                     // If the original binding was a global or nonlocal, and the new binding conflicts within
                     // the current scope, then the new binding is also as the same.
                     Binding {
-                        runtime_usage: existing.runtime_usage,
-                        synthetic_usage: existing.synthetic_usage,
-                        typing_usage: existing.typing_usage,
+                        references: existing.references.clone(),
```

Comment:
Do we have to clone here. It seems we're moving all other data

Reply:
The fields on `existing` need to be cloned, yeah. The fields on `binding` are moved.